src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.SslProtocols.cs

```diff
@@ -155,7 +156,8 @@ public void DisabledProtocols_SetSslProtocols_ThrowsException(SslProtocols disab
                 }
                 using (var client = new HttpClient(handler))
                 {
-                    (await client.GetAsync(url)).Dispose();
+                    Type remoteServerExceptionType = RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? typeof(TaskCanceledException) : typeof(HttpRequestException);
```

Comment:
Is there a way to check that the HttpRequestException is time out? We do not want to filter too much IMO.

Reply:
There's no special timeout exception

Reply:
It has inner exception:
```
System.Net.Http.HttpRequestException : An error occurred while sending the request.
---- System.Net.Http.WinHttpException : The operation timed out
```

I wonder if it would be better to pass a lambda `Exception => bool` instead of type of the exception. The lambda can decide when to mark the exception as "remoteServerException = true".